xarray/plot/plot.py

```diff
@@ -581,8 +580,14 @@ def imshow(x, y, z, ax, **kwargs):
                          'pcolormesh or contour(f)')
 
     # Centering the pixels- Assumes uniform spacing
-    xstep = (x[1] - x[0]) / 2.0
-    ystep = (y[1] - y[0]) / 2.0
+    try:
+        xstep = (x[1] - x[0]) / 2.0
+    except IndexError:
+        xstep = .1
```

Comment:
nit: I find `xstep = .5` more intuitive here

Reply:
ignore my comment: it doesn't matter

Reply:
Either way, it would be good to add a comment here noting that the choice is arbitrary